NLog.Web.AspNetCore/LayoutRenderers/AspNetQueryStringLayoutRenderer.cs

```diff
@@ -0,0 +1,165 @@
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="logEvent"></param>
+        protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
+        {
+            if (this.QueryStringKeys?.Count > 0)
+            {
+                var httpRequest = HttpContextAccessor?.HttpContext?.TryGetRequest();
+
+                if (httpRequest == null)
+                    return;
+
+#if !NETSTANDARD_1plus
+                this.SerializeQueryString(builder, httpRequest.QueryString);
+#else
+                this.SerializeQueryString(builder, httpRequest.Query);
+#endif
+
+            }
+        }
+
+#if !NETSTANDARD_1plus
+        /// <summary>
+        /// To Serialize the QueryString based on the configuration.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="queryStrings"></param>
+        private void SerializeQueryString(StringBuilder builder, NameValueCollection queryStrings)
+        {
+            var includeArrayEndBraces = false;
+
+            if (queryStrings?.Count > 0)
+            {
+                var i = 0;
+                foreach (var configuredKey in this.QueryStringKeys)
+                {
+                    var value = queryStrings[configuredKey];
+
+                    if (!String.IsNullOrEmpty(value))
+                    {
+                        if (i > 0)
+                            builder.Append($",{Environment.NewLine}");
+
+                        switch (this.OutputFormat)
+                        {
+                            case AspNetLayoutOutputFormat.Flat:
+                                builder.Append($"{configuredKey}:{value}");
+                                break;
+                            case AspNetLayoutOutputFormat.Json:
+                                if (!includeArrayEndBraces)
+                                {
+                                    includeArrayEndBraces = true;
+                                    builder.Append(jsonArrayStartBraces);
+                                }
+                                builder.Append($"{jsonElementStartBraces}{doubleQuotes}{configuredKey}{doubleQuotes}:{doubleQuotes}{value}{doubleQuotes}{jsonElementEndBraces}");
+                                break;
+                            default:
+                                break;
+                        }
+                        i++;
+                    }
+                }
+
+                if (includeArrayEndBraces)
+                    builder.Append(jsonArrayEndBraces);
+            }
+        }
+#else
+        /// <summary>
+        /// To Serialize the QueryString based on the configuration.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="queryStrings"></param>
+        private void SerializeQueryString(StringBuilder builder, IQueryCollection queryStrings)
+        {
```

Comment:
I changed this and I m yet to push the code.
On Jun 5, 2016 08:46, "Julian Verdurmen" <notifications@github.com> wrote:

> In NLog.Web.AspNetCore/LayoutRenderers/AspNetQueryStringLayoutRenderer.cs
> <https://github.com/NLog/NLog.Web/pull/54#discussion_r65816870>:
>
> > +                        i++;
> > +                    }
> > +                }
> > +
> > +                if (includeArrayEndBraces)
> > +                    builder.Append(jsonArrayEndBraces);
> > +            }
> > +        }
> > +#else
> > +        /// <summary>
> > +        /// To Serialize the QueryString based on the configuration.
> > +        /// </summary>
> > +        /// <param name="builder"></param>
> > +        /// <param name="queryStrings"></param>
> > +        private void SerializeQueryString(StringBuilder builder, IQueryCollection queryStrings)
> > +        {
>
> Indeed, so you need transforming it to a common type. Only the
> transformation is platform specific.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NLog/NLog.Web/pull/54/files/434738a02003e6b4adc3e9cca81df66b2b7751a5#r65816870>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAmxyQOLTwX_5Jn0kVs2sJLUo9gKJV-aks5qIsUggaJpZM4ITYFh>
> .
>